### DISCUSSION TITLE
Welcome to build Discussions!

### DISCUSSION THREAD

**Discussion by tanaypratap** [neogcamp/build#1] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.
 * **Showcase** your work when you have completed any of the projects.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by AnkitaBagale** [comment]
Hi, 
This is Ankita. I am currently learning JS and React Js. I am new to web dev.
I have created Cash register manager using vanilla JS.

I am happy to share my experience.
1. While coding I realised code is getting repeated so used functions&loops.
2. Firstly code was not generic. If anyone says we also have 200 notes then I would have to make changes at so many places. So I used variables and arrays instead of actual values.

Please check out my app and share your feedback.
https://cashregister-ankitab.netlify.app/
https://github.com/AnkitaBagale/cashRegisterManager

**Reply by desaihetav** [reply]
Hi Ankita, This is Hetav here.

The app looks amazing. Loved the whole UI/UX especially the way you've displayed the output in tabular manner!!

There's just one thing I'd like to point out. You're comparing bill and cash amount in string format and hence it fails in some cases even though the cash is greater than bill. Just convert it to Number and everything will be perfect.

Good work with the app, loved using it!!

**Reply by AnkitaBagale** [reply]
> Hi Ankita, This is Hetav here.
> 
> The app looks amazing. Loved the whole UI/UX especially the way you've displayed the output in tabular manner!!
> 
> There's just one thing I'd like to point out. You're comparing bill and cash amount in string format and hence it fails in some cases even though the cash is greater than bill. Just convert it to Number and everything will be perfect.
> 
> Good work with the app, loved using it!!

Thanks Hetav. That was really helpful. I made the changes and now it's working.

**Comment by arp99** [comment]
Hi! I am Arpan, I tried making the **cash manager** app using vanilla js
Please Do check out and suggest changes :
https://github.com/arp99/Cash-Register-Manager
https://cash-manager.netlify.app/

**Reply by RadhikaRJ** [reply]
Hey Arpan, 

I observed following good aspects of your app:
1. I liked how the result table & cash amount gets erased when the user deletes the input in bill amount. Great work incorporating that!
2. You have disabled cash input till bill input is filled in. Nice way to have the user focus on entering one input at a time.
3. Bill input & Cash Amount is strictly numeric & does not register any alphabets keyed in by the user. Well handled user input!

Suggestions: 
1. You might want to check on how to have the hover effect enabled when user is using tab key to navigate, so that the button  gets highlighted.
2. Check out these color pallets, hope you find some inspiration here!
https://www.canva.com/colors/color-palettes/

**Reply by arp99** [reply]
Ok then I need to add tabindex for the HTML elements. 
Ok thanks I'll surely implement the tab navigation. 
*Thanks for ur suggestion*

**Reply by RadhikaRJ** [reply]
I believe the tab does work for your button, so the tab navigation need not be explicitly implemented. Its just not getting highlighted when the control of tab is at your button.  The :active & :focus in CSS should be an easier way of getting it done. Do let me know how tab navigation implementation goes for you.

**Reply by arp99** [reply]
Already done you may check it once more

**Comment by prerana1821** [comment]
Hello! I'm Prerana,
I'm currently learning Javascript.
I have created the cash register manager & the lucky-birthday app using vanilla js.
Do check it out, suggestions & feedbacks are welcomed!

Cash Register Manager
https://cash-register.prerananawar.repl.co/
https://github.com/prerana1821/build-basics/tree/main/cash-register

Lucky Birthday
https://lucky-birthday.prerananawar.repl.co/
https://github.com/prerana1821/build-basics/tree/main/is-your-birthday-lucky

**Comment by RadhikaRJ** [comment]
Hi Friends,
Radhika here..

I am new to web development. Using Vanilla JS, CSS & HTML I have created a Lucky Birthday Identifier Application.
I have learnt loads from UI & feature perspective in this application. 

Some of my learnings include:
1. Better button designing & hover effect
2. Improved Text Styling
3. Layout & color combinations using div tag
4. Switching displayed images based on result 
5. Download document feature for privacy notice & removing the alert from UI when user clicks on "x"

Link to application: https://rj-luckybirthday.netlify.app/
Link to source code: https://github.com/RadhikaRJ/neogcamp_luckyBirthday

update 28/12/2020: 
Palindrome Application was challenging to implement. Modular division of tasks & reusability of code was something that helped me get it done. Apart from logic development, I majorly learnt following:
1. Date() library & it's implementation
2. Experienced how JavaScript assumes a data type & the need to cast the data to get appropriate results.
3. Embedding GIFs in UI

Link to application: https://rj-pallindrome-date.netlify.app/
Link to source: https://github.com/RadhikaRJ/neogcamp_pallindrome_birthday

Requesting feedback, especially on how I can improve on Designing the UI.

**Comment by amansethi00** [comment]
Hi! I am Aman, I tried making the cash manager app and is-your-birthday-lucky app using vanilla js with minimal styling.

Cash Register App
Please Do check out and suggest changes: https://5apps.netlify.app/basics/cash-register-app/index.html
Link to source code: https://github.com/amansethi00/build/tree/main/basics/cash-register-app

Is your birthday lucky?
Please Do check out and suggest changes: https://5apps.netlify.app/basics/is-your-birthday-lucky/index.html
Link to source code: https://github.com/amansethi00/build/tree/main/basics/is-your-birthday-lucky

**Reply by RadhikaRJ** [reply]
Hey Aman,

Good work on both the apps! Especially the controlled display of cash & bill inputs looks great!

Some suggestions:
The selected colors of Lucky birthday seem a bit odd together to me. Try out some contrasting set of shades/lighter tones..
For the lucky birthday app, try showing a different image based on output( yes lucky,/no not lucky)
Using the appendChild() function of HTML element can help you to implement that.

2/5 done, way to go :+1:

**Reply by amansethi00** [reply]
> Hey Aman,
> 
> Good work on both the apps! Especially the controlled display of cash & bill inputs looks great!
> 
> Some suggestions:
> The selected colors of Lucky birthday seem a bit odd together to me. Try out some contrasting set of shades/lighter tones..
> For the lucky birthday app, try showing a different image based on output( yes lucky,/no not lucky)
> Using the appendChild() function of HTML element can help you to implement that.
> 
> 2/5 done, way to go 👍

Thank you Radhika, I will update my lucky birthday app according to changes suggested by you.

**Comment by puneetsingh99** [comment]
Hii, I am Puneet. I have created the Cash Register Manager app using vanilla JS.
I have implemented the app in such a way that it doesn't require to click any buttons to get the output.

The navigation flow is:
1. Type the amount on the "Enter bill" input and **press enter** the "Enter cash" will be disabled until the bill is entered. The focus will shift to the "Enter cash" input.
2. Type the amount given by the user and **press enter**.
3. The appropriate output will be displayed and the focus will again come back to the "Enter bill" input and the same cycle can be continued without resetting the app.

Please give feedbacks to help me improve.

live app: https://cash-register-manager-bypuneet.netlify.app/
GitHub repo: https://github.com/puneetsingh99/cash-register-manager
